src/models/identity-provider-role.ts

```diff
@@ -0,0 +1,21 @@
+import Joi from 'joi';
+
+interface IdentityProviderRole {
+  hashId: string;
+  identityProviderHashId: string;
+  clientHashId?: string;
+  supplierHashId?: string;
+  role: string;
+  rights: string[];
+}
+
+const schema = (): Joi.ObjectSchema => Joi.object().keys({
+  hashId: Joi.string().required().example('wasd2'),
+  identityProviderHashId: Joi.string().required().example('ds23d'),
+  clientHashId: Joi.string().example('as31db'),
+  supplierHashId: Joi.string().example('a2f1da'),
+  role: Joi.string().required(),
+  rights: Joi.array().items(Joi.string().example('READ')).required(),
```

Comment:
**Question**

Couldn't we have a more strongly typed `rights`/`roles` ?

You don't have to do anything here, but I would expect that we have a shared schema and (ts) type for these.

Reply:
For roles -> no they will be defined by stedin and I definitely do not want to change our sdk if they decide to add a new role or change the spelling of one.
For rights yes but do you really want to? It is currently consistent with all other places where we define rights like https://github.com/withthegrid/platform-sdk/blob/22058adabb65be75b3db90734c0b3fb8f2fad6d1/src/models/user.ts#L14 And if we promise a certain number of acceptable values then introducing a new right would become a breaking change in the sdk.

Reply:
> And if we promise a certain number of acceptable values then introducing a new right would become a breaking change in the sdk

Very good point. I could argue that it doesn't have to be, but I don't have anything to add to verify this argument.
And I could also say that in some cases, having a change of rights **should be** a breaking change.

But none of the above is supported by proofs from me, and I like the first point: "It is currently consistent with all other places where we define rights"

Thanks for the answer 🙏